ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala

```diff
@@ -177,4 +186,95 @@ final class ConfigManagementServiceIT extends LedgerTestSuite {
     }
   })
 
+  test(
+    "DuplicateSubmissionIdTwiceCorrect",
+    "Changing config twice with the same submissionId succeeds only on two participants",
+    allocate(NoParties, NoParties),
+    runConcurrently = false,
+  )(implicit ec => { case Participants(Participant(alpha), Participant(beta)) =>
+    // Multiple config changes should never cause duplicates. Participant adds extra entropy to the
+    // submission id to ensure client does not inadvertently cause problems by poor selection
+    // of submission ids.
+    for {
+      (req1, req2) <- generateRequest(alpha).map(r =>
+        r -> r
+          .update(_.configurationGeneration := r.configurationGeneration + 1)
+      )
+      _ <- ignoreNotAuthorized(alpha.setTimeModel(req1))
+      _ <- ignoreNotAuthorized(beta.setTimeModel(req2))
+    } yield ()
+
+  })
+
+  test(
+    "DuplicateSubmissionIdFirstIncorrect",
+    "Changing config twice with the same submissionId fails for first incorrect submission",
+    allocate(NoParties, NoParties),
+    runConcurrently = false,
+  )(implicit ec => { case Participants(Participant(alpha), Participant(beta)) =>
+    for {
+      (good, bad) <- generateRequest(alpha).map(r =>
+        r -> r
+          .update(_.configurationGeneration := r.configurationGeneration + 2)
+      )
+      failure <- ignoreNotAuthorized(beta.setTimeModel(bad))
+        .mustFail("Mismatching configuration generation")
+      _ <- alpha.setTimeModel(good)
+    } yield {
+      assertGrpcError(failure, Status.Code.INVALID_ARGUMENT, "Mismatching configuration generation")
+    }
+  })
+
+  test(
+    "DuplicateSubmissionIdSecondIncorrect",
+    "Changing config twice with the same submissionId fails for first incorrect submission",
```

Comment:
The description is the same as for the previous test. 
Looking better at them, they look highly similar. Did you consider only writing one test with the following order that would assert both cases:
```
_ <- alpha.setTimeModel(good_1)
failure <- alpha.setTimeModed(bad)
_ <- alpha.setTimeModel(good_2)
} yield ...
```

Reply:
Done